snuba/admin/static/rpc_endpoints/index.tsx

```diff
@@ -1,240 +1,304 @@
 import React, { useState, useCallback, useEffect } from 'react';
-import { Select, Button, Code, Space, Textarea, Accordion, createStyles, Loader, Checkbox, Text, Table } from '@mantine/core';
+import { Select, Button, Code, Space, Textarea, Accordion, createStyles, Loader, Checkbox, Text, Table, Switch } from '@mantine/core';
 import useApi from 'SnubaAdmin/api_client';
+import { format_trace_log } from 'SnubaAdmin/rpc_endpoints/trace_formatter';
+import { QueryInfo } from 'SnubaAdmin/rpc_endpoints/types';
 
 const DEBUG_SUPPORTED_VERSIONS = ['v1'];
 
 function RpcEndpoints() {
-    const api = useApi();
-    const [endpoints, setEndpoints] = useState<Array<{ name: string, version: string }>>([]);
-    const [selectedEndpoint, setSelectedEndpoint] = useState<string | null>(null);
-    const [selectedVersion, setSelectedVersion] = useState<string | null>(null);
-    const [requestBody, setRequestBody] = useState('');
-    const [response, setResponse] = useState<any | null>(null);
-    const exampleRequestTemplates: Record<string, Record<string, any>> = require('SnubaAdmin/rpc_endpoints/exampleRequestTemplates.json');
-    const [accordionOpened, setAccordionOpened] = useState(false);
-    const [isLoading, setIsLoading] = useState(false);
-    const [debugMode, setDebugMode] = useState(false);
+  const api = useApi();
+  const [endpoints, setEndpoints] = useState<Array<{ name: string, version: string }>>([]);
+  const [selectedEndpoint, setSelectedEndpoint] = useState<string | null>(null);
+  const [selectedVersion, setSelectedVersion] = useState<string | null>(null);
+  const [requestBody, setRequestBody] = useState('');
+  const [response, setResponse] = useState<any | null>(null);
+  const exampleRequestTemplates: Record<string, Record<string, any>> = require('SnubaAdmin/rpc_endpoints/exampleRequestTemplates.json');
+  const [accordionOpened, setAccordionOpened] = useState(false);
```

Comment:
Naming too ambiguous, what does this accordion do?

Reply:
The accordion in this code is a collapsible UI component  that shows/hides the "Example Request Payload" section.

Reply:
right so name it after what it does, `exampleRequestAccordionOpened`